Firmware/mmu2.cpp

```diff
@@ -676,25 +681,26 @@ void MMU2::OnMMUProgressMsg(ProgressCode pc){
                 switch ( WhereIsFilament() )
                 {
                 case FilamentState::AT_FSENSOR:
-                    // fsensor triggered, stop moving the extruder
+                    // fsensor triggered, finish FeedingToBondtech state
                     loadFilamentStarted = false;
-                    // TODO: continue to ProgressCode::FeedingToNozzle?
+                    // After the MMU knows the FSENSOR is triggered it will:
+                    // 1. Push the filament by additional 30mm (see fsensorToNozzle)
+                    // 2. Disengage the idler and push another 5mm.
+                    current_position[E_AXIS] += 30.0f + 5.0f;
+                    plan_buffer_line_curposXYZE(MMU2_LOAD_TO_NOZZLE_FEED_RATE);
                     break;
                 case FilamentState::NOT_PRESENT:
                     // fsensor not triggered, continue moving extruder
-                    // TODO: Verify what's the best speed here?
-                    current_position[E_AXIS] += MMU2_LOAD_TO_NOZZLE_FEED_RATE;
+                    current_position[E_AXIS] += 5.0f;
                     plan_buffer_line_curposXYZE(MMU2_LOAD_TO_NOZZLE_FEED_RATE);
-                    st_synchronize(); // Wait for the steps to be done, otherwise the moves will just add up
+                    st_synchronize(); // Wait for the steps to be done so the moves don't pile up
```

Comment:
:thinking: so this is the barrier preventing from having too many 5mm stages ... I don't like the fact, that st_synchronize will block the OnMMUProgressMsg handler until the 5mm move is actually performed.

Reply:
Yup I noticed `MMU2::OnMMUProgressMsg` is called multiple times before a single move is completed and thus they could pile up. `st_synchronize()` solved that.

Reply:
@DRracer one idea I have:
Use same function `blocks_queued()` as `st_synchronize()` but just use an `if` statement instead of `while()` such that it is not blocking. But your comment above about the frequency of queries is still valid. 
```c
                case FilamentState::NOT_PRESENT:
                    // fsensor not triggered, continue moving extruder
                    if (!blocks_queued())
                    {
                        current_position[E_AXIS] += 5.0f;
                        plan_buffer_line_curposXYZE(MMU2_LOAD_TO_NOZZLE_FEED_RATE);
                    }
```

Reply:
I added a commit with this change above. Tested it on my printer.